### ISSUE TITLE
[R package] Confusion of "pred2 should be identical to pred"

### ISSUE THREAD

**Issue by zheguzai100** [Microsoft/LightGBM#674]
@henry0312 
@guolinke
running `basic_walkthrough.R`
`sum(abs(pred2-pred))= 2.00211185941354`
in my mind, 'be identical to' means 0, or at least infinitily be close to 0, why 2.002
and how to judge a value whether means 'idnetical to each other'

##### -------------------------------
`require(lightgbm)`
`require(methods)`
`require(Matrix)`
#### In this example, we are aiming to predict whether a mushroom is edible
`data(agaricus.train, package = "lightgbm")`
`data(agaricus.test, package = "lightgbm")`
`train <- agaricus.train`
`test <- agaricus.test`
`bst <- lightgbm(data = train$data,
                label = train$label,
                num_leaves = 4,
                learning_rate = 1,
                nrounds = 2,
                objective = "binary")`
`pred <- predict(bst, test$data)`
`err <- mean(as.numeric(pred > 0.5) != test$label)`
`print(paste("test-error=", err))`
`lgb.save(bst, "I:/rwork/lightGBM/lightgbm.model")`
#### Load binary model to R
`bst2 <- lgb.load("I:/rwork/lightGBM/lightgbm.model")`
`pred2 <- predict(bst2, test$data)`
#### pred2 should be identical to pred
`print(paste("sum(abs(pred2-pred))=", sum(abs(pred2 - pred))))`
#### [1] "sum(abs(pred2-pred))= 2.00211185941354"

**Comment by zheguzai100** [comment]
`print(paste("sum(abs(pred2-pred))=", sum(abs(pred2 - pred))))`
`[1] "sum(abs(pred2-pred))= 2.00211185941354"`
`print(paste("mean(abs(pred2-pred))=", mean(abs(pred2 - pred))))`
`[1] "mean(abs(pred2-pred))= 0.0012427758283138"`

**Comment by guolinke** [comment]
@zheguzai100 Thanks, fixed in the latest commit.

**Comment by zheguzai100** [comment]
@guolinke 
demo of boost_from_prediction:
`param <- list(num_leaves = 4,
              learning_rate = 1,
              nthread = 2,
              silent = 1,
              objective = "binary")`
`bst <- lgb.train(params=param, data=dtrain,
                 nrounds = 1, valids = valids)`
#### [LightGBM] [Fatal] Unknown parameter: silent
#### Error in lgb.call("LGBM_DatasetCreateFromCSC_R", ret = handle, private$raw_data@p,  : 
 ####  api error: Unknown parameter: silent


`silent=1` means `verbose = 0` ?

**Comment by guolinke** [comment]
yeah, it should be ```verbose=0``` .
